### PR TITLE
changed the tree structure of the app/ useInput, etc.

### PR DESCRIPTION
**Pull request by BelohvostovAlex**

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import { ButtonProps } from './interfaces'
+
+import './formButton.scss'
+
+export const Button: React.FC<ButtonProps> = ({
+  btnText,
+  btnType,
+  btnClass,
+  onSubmit,
```
Comment (igorTT) [inline src/components/Button/Button.tsx]: Replace with `onClick` it can be cancel button

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+
+import { ButtonProps } from './interfaces'
+
+import './formButton.scss'
```
Comment (igorTT) [inline src/components/Button/Button.tsx]: Use one type of quotes. Prefer single-quote

File: src/components/Button/ButtonContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+
+import { Button } from "./Button";
+import { ButtonContainerProps } from "./interfaces";
+
+export const ButtonContainer: React.FC<ButtonContainerProps> = ({
+  btnClass,
+  btnType,
+  onSub,
```
Comment (igorTT) [inline src/components/Button/ButtonContainer.tsx]: Don't save the place on the screen. Use readable names. onSub- on subscribe?

File: src/components/Button/ButtonContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  };
+  return (
+    <Button
+      btnClass={btnClass}
+      btnType={btnType}
+      onSubmit={onSubmitForm}
+      btnText={btnText}
+    />
+  );
+};
```
Comment (igorTT) [inline src/components/Button/ButtonContainer.tsx]: Use a simple button. Don't try to spread the logic. I don't see the reason to preventDefault. If you need to do it- it'll be easier to do it on some upper level

File: src/components/Form/Form.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import React from "react";
+
+import { ButtonContainer } from "../Button";
```
Comment (igorTT) [inline src/components/Form/Form.tsx]: Replace with a simple button

File: src/components/Form/FormContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import React, { useState } from "react";
+import axios from "axios";
+
+import { Form } from "./Form";
+import { IUser } from "../../models/IUser";
+import { useInput } from "../../hooks/useInput";
+
+export const FormContainer = () => {
+  const username = useInput("");
+  const password = useInput("");
```
Comment (igorTT) [inline src/components/Form/FormContainer.tsx]: Replace with
```javascript
  const [username, setUserame] = useInput();
  const [password, setPassword] = useInput();

File: src/components/Form/FormContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+  const onSubmit = async () => {
+    try {
+      const response = await axios.post<IUser>("http://localhost:4000/api", {
+        username: username.value,
+        password: password.value,
+      });
+      setOutput(response.data);
+      username.setValue("");
+      password.setValue("");
```
Comment (igorTT) [inline src/components/Form/FormContainer.tsx]: Pass an empty parameter to the handler. It'll be more common logic.

File: src/hooks/useInput.tsx [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import React, { useState } from 'react'
+
+export const useInput = (initialValue: string) => {
+  const [value, setValue] = useState<string>(initialValue)
+
+  const handleValue = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value)
+  }
+
+  return {value, setValue, handleValue}
```
Comment (igorTT) [inline src/hooks/useInput.tsx]: You don't need to use setValue on the lower level

**Review by igorTT — CHANGES_REQUESTED** [review]

File: src/components/Button/Button.tsx (hunk 2 above)
Comment (BelohvostovAlex) [reply src/components/Button/Button.tsx]: done

File: src/components/Button/ButtonContainer.tsx (hunk 4 above)
Comment (BelohvostovAlex) [reply src/components/Button/ButtonContainer.tsx]: done

File: src/components/Form/FormContainer.tsx (hunk 6 above)
Comment (BelohvostovAlex) [reply src/components/Form/FormContainer.tsx]: done

**Review by igorTT — APPROVED** [review]

**Pull request merged**